### ISSUE TITLE
Node built-ins: Cannot find name 'Buffer'

### ISSUE THREAD

**Issue by larssn** [angular/angular-cli#6880]
Not sure this is a bug, but read on if you please.

We're coming from Ionic, and are converting our existing codebase to pure Angular.
In Ionic, using the Node built-in `Buffer`, was allowed.

So having something like:
```
for (let i = 32; i <= 255; i++) {
   this.print(Buffer.from([i]));
}
```
.. worked; however webpack throws the error: `Cannot find name 'Buffer'.`, and my Google Fu has failed me.

I've tried the Buffer polyfill: `npm install buffer` - no luck.
I've tried the node types `npm install @types/node --save-dev` - no luck.

I'm not sure how to proceed, and why it worked in Ionic-land in the first place.

So bug? No idea, but it's my understanding that Buffer should be a default shim in Webpack (as per https://webpack.js.org/configuration/node/#node-buffer)

### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.
@angular/cli: 1.1.3 (e)
node: 8.1.3
os: darwin x64
@angular/animations: 4.2.5
@angular/common: 4.2.5
@angular/compiler: 4.2.5
@angular/core: 4.2.5
@angular/forms: 4.2.5
@angular/http: 4.2.5
@angular/platform-browser: 4.2.5
@angular/platform-browser-dynamic: 4.2.5
@angular/router: 4.2.5
@angular/cli: 1.1.3
@angular/compiler-cli: 4.2.5
@angular/language-service: 4.2.5

### Simple Repro steps.
`ng new buffer-test`.
`npm install @types/node --save-dev`.
`npm install buffer --save`.
Create a variable of type `Buffer` in `app.component.ts`:
```
export class AppComponent {
  title = 'app';
  test: Buffer;
}
```
`ng build`.

Error: `src/app/app.component.ts (10,9): Cannot find name 'Buffer'.`

**Comment by Brocco** [comment]
I believe you're just missing an `import`
```
import {Buffer} from 'buffer';
```
Please try and report back.

**Comment by larssn** [comment]
That seems painfully obvious in retrospect: It works.

Strange it wasn't required in Ionic.

Closing.

**Comment by frankyso** [comment]
how do u solve this problem @larssn ?

**Comment by larssn** [comment]
This was ages ago, so forgive me if this is wrong.

1. First, make sure you have the node types: `npm i -D @types/node`.

2. Last I checked, Angular did not use any of the node built-ins, so you do not get a polyfill for `Buffer` out of the box. You might need to modify your Angular webpack.config.js to include it (how you do that varies, but last I messed around with it, I used an injection plugin. I suggest you Google around if there still is no official solution).

3. Using whatever injection method you figured out above, add "buffer" to webpack.config, see: https://webpack.js.org/configuration/node/#node . Behind the scenes, it uses the following library: https://github.com/webpack/node-libs-browser

I think that should do it.
